EXAMPLES.md

```diff
@@ -0,0 +1,202 @@
+# Examples
+
+- [Request Options](#request-options)
+- [Pagination](#pagination)
+  - [Page based pagination](#page-based-pagination)
+  - [Checkpoint pagination](#checkpoint-pagination)
+- [Custom User Structs](#providing-a-custom-user-struct)
+
+## Request Options
+
+Fine-grained configuration can be provided on a per-request basis to enhance the request with specific query params, headers or to pass it a custom context.
+
+```go
+// Example
+userGrants, err := auth0API.Grant.List(
+	management.Context(ctx)
+	management.Header("MySpecialHeader","MySpecialHeaderValue")
+    management.Parameter("user_id", "someUserID"),
+    management.Parameter("client", "someClientID"),
+    management.Parameter("audience", "someAPIAudience"),
+)
+
+// Other helper funcs.
```

Comment:
```suggestion
// Other helper funcs
```

Reply:
Would it make sense to also state that not all the endpoints accept these? And it's best to check against the Auth0 API management docs to see which ones do?

Reply:
Yes, it can lead to a frustrating experience otherwise.

Reply:
Added a note above this codeblock stating something to this effect